website/docs/reference/dbt-jinja-functions/builtins.md

```diff
@@ -15,9 +15,43 @@ The `builtins` variable is a dictionary containing the following keys:
 
 ## Usage
 
-The following macro overrides the `ref` method available in the model compilation context to return a [Relation](/reference/dbt-classes#relation) with the database name overriden to `dev`.
+<VersionBlock firstVersion="1.6">
+  
+From dbt v1.6 and higher, use the following macro to extract user-provided arguments, including <code>version</code>, and call the <code>builtins.ref()</code> function with either a single <code>modelname</code> argument or both <code>packagename</code> and <code>modelname</code> arguments, based on the number of positional arguments in <code>varargs</code>:
 
+<br /><br />
+
+  
+```jinja2
+jinja2.ext.do
+
+{% macro ref() %}
+-- extract user-provided positional and keyword arguments
+  {% set version = kwargs.get('version') %}
+  {% set packagename = none %}
+  {%- if (varargs | length) == 1 -%}
+    {% set modelname = varargs[0] %}
+{%- else -%}
+    {% set packagename = varargs[0] %}
+    {% set modelname = varargs[1] %}
+{% endif %}
+-- call builtins.ref based on provided positional arguments
+{% if packagename is not none %}
+    {% do return(builtins.ref(packagename, modelname, version=version)) %}
+{% else %}
+    {% do return(builtins.ref(modelname, version=version)) %}
+{% endif %}
+
+{% endmacro %}
 ```
+</VersionBlock>
+
+<VersionBlock lastVersion="1.5">
+
+From dbt v1.5 and lower, use the following macro to override the `ref` method available in the model compilation context to return a [Relation](/reference/dbt-classes#relation) with the database name overriden to `dev`:
```

Comment:
*dbt v1.4 and lower

I think this example macro just _never_ supported two-argument `ref`. I guess it was a lot less common in earlier versions, but still! Some of the complexity above arrived with the addition of `version`, but some of it really should have been there all along.